### PR TITLE
Cu 27gypxm i os develop the login page UI marty ulrich

### PR DESCRIPTION
**Pull request by martyu**

### CODE REVIEW

**Comment by newmskywalker** [comment]
Task linked: [CU-27gypxm iOS: Develop the login page UI](https://app.clickup.com/t/27gypxm)

File: shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package io.projectnewm.login
+
+class LoginFieldValidator {
+    public companion object {
+        public fun validate(email: String, password: String): Boolean {
```
Comment (andymiller813) [inline shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt]: No need for the companion object - 

```suggestion
object LoginFieldValidator {
    public fun validate(email: String, password: String): Boolean {
        .....
```

File: shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            return isValidEmail(email) && isValidPassword(password)
+        }
+
+        private fun isValidEmail(email: CharSequence?): Boolean {
```
Comment (andymiller813) [inline shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt]: you switched from non-null String in the `validate` function to nullable CharSequence here and nullable String in `isValidPassword`. Stick with either `CharSequence?` or `String?` all the way through (probably CharSequence? I think)

File: shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package io.projectnewm.login
+
+class LoginFieldValidator {
```
Comment (andymiller813) [inline shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt]: 👍

File: shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                        "(" +
+                        "\\." +
+                        "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,25}" +
+                        ")+"
+            )
+            return email?.matches(emailAddressRegex) ?: false
+        }
+
+        private fun isValidPassword(password: String?): Boolean {
+            val password = password?.let { it } ?: return false
```
Comment (andymiller813) [inline shared/src/commonMain/kotlin/io/projectnewm/login/LoginFieldValidator.kt]: It's cleanest to just do 
```suggestion
           return if (password == null || password.isBlank()) false else true
```

**Review by andymiller813 — APPROVED** [review]

**Pull request merged**